pkgs/applications/video/kodi/addons/inputstream-adaptive/default.nix

```diff
@@ -1,16 +1,29 @@
 { stdenv, lib, rel, addonDir, buildKodiBinaryAddon, fetchFromGitHub, expat, glib, nspr, nss, gtest }:
+let
+  bento4 = fetchFromGitHub {
```

Comment:
Why not use our https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/video/bento4/default.nix ?

Reply:
- our `bento4` doesn't resemble what upstream distributes so this isn't currently an option, though i did some work on that and could fix it ...if i find motivation to do so
- even if i were to fix the above point `kodi` is nefarious for vendoring their own modified dependencies for years and years on end - i took a look and what `kodi` has for `bento4` isn't even close to the upstream release :disappointed:

Reply:
The proper fix (and I'm playing fast and loose with "proper" here) is to override `src` on the nixpkgs `bento4` package and build that normally - then have the plugin build with an external bento4 and pass it the custom one. This allows us to cache the bento4 package and benefit for what custom logic we currently have in nixpkgs's bento4.

Reply:
> * our `bento4` doesn't resemble what upstream distributes so this isn't currently an option, though i did some work on that and could fix it ...if i find motivation to do so

I guess I need to emphasize this more. Our `bento4` package looks **nothing** like what upstream (or `kodi`l provides. The entire directory structure is different and there are almost no overlapping files. The way we build `bento4` is almost completely incompatible (wrong?). There is nothing of value in our `bento4` package for `kodi`.

If course of I go ahead and "fix" our `bento4` then what you said is entirely accurate, which is why you said it 👍

I'll try to find some motivation to finish my `bento4` work (very close) and then make a PR... at some point.

On that note I would also like to clean up how we build all these `kodi` vendored packages and do the same for the rest of the dependencies we don't do this for yet.

Let's leave this comment unresolved as an indicator I'll come back to this at some point in the future after this PR is merged.